test/consumer.integration.spec.ts

```diff
@@ -28,6 +28,70 @@ describe('FFI integration test for the HTTP Consumer API', function () {
     value,
   });
 
+  describe('with matching rules', function () {
+    beforeEach(function () {
+      pact = makeConsumerPact(
+        'matcher-consumer',
+        'matcher-provider',
+        FfiSpecificationVersion['SPECIFICATION_VERSION_V3']
+      );
+
+      const interaction = pact.newInteraction('matcher test');
+      interaction.uponReceiving('a write request');
+      interaction.given('a request to write test.txt');
```

Comment:
I was confused by the matching rules because I thought it was a plain text file (`.txt` extension) being returned as a multipart response body. But actually, it's just a JSON body, with a path to a file in the response.

That might also be because I was expecting the use case here to be a multiple body (based on the use case driving this PR) - which I think might actually be a better demonstration here.

I think the example would be clearer if you didn't refer to a file though, and shows the use of the matcher on some generic response body.

We can use Pact JS to demonstrate the matching rules on multipart.

Reply:
changed to a plain json from the file like test.

Reply:
initially wanted to add the multipart test here, but as the pact-js lib wasn't installed here seeing as this is its dependency i changed the test to be slighly simpler as ill add that test to pact-js examples instead.